### DISCUSSION TITLE
C++ LLaMA 2 example? Xcode build instructions?

### DISCUSSION THREAD

**Discussion by douglasdew** [ml-explore/mlx#35] [General]
Please provide a C++ LLaMA 2 example. And please provide Xcode build instructions.

**Comment by dougdew64** [comment]
With a bit of grepping, I was able to determine the settings to use Xcode to successfully build the c++ tutorial example code as a command line app which links to the mlx library which is built from the Makefile which is generated for mlx by cmake.

However, my goal is to use Xcode's debugger to step into the mlx code from the c++ tutorial code. So, I'd like to add all of the mlx source files to my Xcode project and then build all source files as part of that project.

**Comment by dougdew64** [comment]
I created a workspace with two projects: an external build system (Makefile) project for the mlx library, and a second project for tutorial.cpp. Probably I can figure out the rest of the details from here.

**Comment by dougdew64** [comment]
Now it's time to spend some hours in the Xcode debugger learning how mlx works.

![image](https://github.com/ml-explore/mlx/assets/98898390/1c63973d-b1af-4841-b5dd-68a5c93d349e)

**Reply by dougdew64** [reply]
Oops. All that I actually gained debugger access to are the inlines in the mlx header files.

**Reply by dougdew64** [reply]
And, there are little glitches such as Xcode not permitting me to declare that my client project (tutorial) depends upon my mlx project. That seems to be related to the mlx project being an external build system project, as I've had no problem in the past declaring dependencies upon projects which are built within Xcode.

**Comment by alikhan-tech** [comment]
To build the C++ LLaMA 2 example in Xcode, you'll need to follow these general steps:

1. **Setting up Xcode Project:**
   - Open Xcode and create a new C++ project or import the LLaMA 2 example into an existing project.
   - Ensure your project settings are configured to support C++ compilation.

2. **Include LLaMA 2 Example Code:**
   - Add the LLaMA 2 example code files to your Xcode project. Make sure to retain the directory structure if it exists.

3. **Configuration and Dependencies:**
   - Check if the example code relies on any external libraries or dependencies. Make sure these are properly included or installed in your project.

4. **Build and Compile:**
   - Navigate to the target or main file that contains the `main()` function for the LLaMA 2 example.
   - Configure any necessary build settings within Xcode for your C++ project.
   - Build and compile the project. You might need to resolve any errors or warnings that arise during compilation.

5. **Running the Example:**
   - Once successfully compiled, execute the compiled binary or run the project within Xcode to test the LLaMA 2 example.

Always refer to the specific documentation or instructions provided with the LLaMA 2 example for any additional configuration steps or peculiarities related to Xcode or C++ compilation. These steps provide a general guideline for integrating and building C++ code within an Xcode project.

**Reply by N3mes1s** [reply]
lol is this generated by llm? Feels like the classic chatgpt steps 👯 apart from the reference to LLaMA2 that it is non-sense.
